dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/db/DatabaseTaskStore.java

```diff
@@ -276,13 +279,46 @@ public boolean createProperty(String name, String value) throws Exception {
             em.persist(property);
             em.flush();
         } catch (EntityExistsException x) {
-            return false;
+            /*
+             * FIXME JPA Spec says em.persist should throw EntityExistsException if entity already exists.
+             * EclipseLink implementation throws PersistanceException during the flush operation.
+             * If EclipseLink changes behavior, this exception will notify us that we need to reactor
+             * this catch to return false, and subsequent PersistenceException can remove extra checks.
+             */
+            throw new UnsupportedOperationException("EntityExistsException thrown during persist operation. "
```

Comment:
Is it possible for a user to use a different JPA implementation for concurrent persistence or do we always use the eclipselink impl?

Reply:
Persistence service currently only allows the use of EclipseLink, but I'm glad you thought of that possibility, because we will want to take it into account.
Given that the error will be the first thing the JPA team sees when they make changes in this area -- either to enhance EclipseLink or to support a different JPA persistence provider -- let's be more explicit in the exception message.  Maybe something like: "EntityExistsException codepath indicates that additional function has been added. See comments in code for how to handle."

Other than that, some very minor typo fixes are:
PersistanceException -> PersistenceException
reactor --> refactor